### ISSUE TITLE
Clarifying data collection to end users

### ISSUE THREAD

**Issue by kevinrobinson** [IBM/taxinomitis#172]
I read https://cloud.ibm.com/docs/services/assistant?topic=assistant-information-security#information-security as suggesting that products using IBM services own responsibility for ethical use, and legal liability for ensuring compliance with local laws and regulations (eg, GDPR).

I also see pretty explicit guidance on personally identifiable information that doesn't seem like it's presented to end users:

<img width="644" alt="Screen Shot 2019-03-15 at 11 20 08 AM" src="https://user-images.githubusercontent.com/1056957/54443219-9075c280-4716-11e9-8767-f6a82f0fe4c4.png">

To me, helping young people understand these responsibilities seems super important for teaching them how to build AI systems ethically.  Would you be open to a pull request that tried to do this?

**Comment by kevinrobinson** [comment]
oops, thinks I put this on the wrong issue earlier, sorry!

Also, the help awesomely describes the site's use of things like the analytics and error reporting services.

<img width="1059" alt="Screen Shot 2019-03-15 at 11 46 05 AM" src="https://user-images.githubusercontent.com/1056957/54443869-f151ca80-4717-11e9-9622-be2d690e2474.png">

Teachers or parents may read this copy and have the expectation that this is the exhaustive list of where the data that they submit is stored.  But it doesn't mention anything about sending data to IBM services, and that data storage, so it seems like maybe this should be updated to include that as well.

What do you think makes sense?

**Comment by dalelane** [comment]
Yeah, I'd be happy with a pull request that improved the wording. 

The English version of the text is here:
https://github.com/IBM/taxinomitis/blob/adaec31ee07a9dc336334315e56fad4aa5884d05/public/languages/en.json#L1001-L1010

And is displayed here:
https://github.com/IBM/taxinomitis/blob/adaec31ee07a9dc336334315e56fad4aa5884d05/public/components/help/help.html#L362-L381

**Comment by kevinrobinson** [comment]
@dalelane awesome!  I opened https://github.com/IBM/taxinomitis/pull/173 as a first step, thanks for your help! 👍 

I also thought it might be good to put a notice and link with guidelines directly into the point of the app where folks are adding training data, what do you think?  That way folks can realize where the data is going, even if they don't go dig into the fine print like I did.

So on the screen like this...
<img width="1190" alt="Screen Shot 2019-03-20 at 5 08 55 PM" src="https://user-images.githubusercontent.com/1056957/54719968-96174200-4b34-11e9-87ef-3e53f3e77313.png">

...one approach might be to add something like this:
<img width="540" alt="Screen Shot 2019-03-20 at 5 28 24 PM" src="https://user-images.githubusercontent.com/1056957/54720357-96640d00-4b35-11e9-9e26-ae31798a1903.png">

The idea is that this might help prevent any surprises, especially for young children who are new to ML and services.  Thanks for listening, and I'm happy to help out with this too, or with other ideas you have.  I didn't do anything in this PR, and wanted to see what you thought first.  👍 Thanks!

**Comment by dalelane** [comment]
I've made an effort to keep the UI for kids as clean, simple and uncluttered as possible. The majority of the site users, as far as I know, are primary school age. I think this sort of legalese warning is unlikely to be useful or helpful to a 7 year old. At worst, it'll confuse them. At best, they'll likely ignore it. My (admittedly untested) assumption is that a message like this won't really be an effective way to address the issue here. 

The simplest approach would be to separate this out for teacher and student users. Give the teacher/parent users all the detailed info (links to more info, explain what is happening, explain the implications, etc.) and keep it out of the student / training UI. I'm very comfortable putting any information in the hands of teachers/parents and letting them decide what to do about it, and what is appropriate to tell their children/students. 

The more nuanced approach would be to also add something to the training/student UI, but make it much more child-friendly. If anything is going in the training UI, it needs to be something that would make sense to a young child. That needs more thought about where it should go, the sort of language that should be used, how it should be explained, how it should be presented, etc.

**Comment by kevinrobinson** [comment]
@dalelane This is awesome, thanks for sharing all this! ❤️ 👍 

Yeah, I think your question gets right to the heart of this - as we make awesome new ways for young children to make their own things with computing, and use more powerful tools like third-party services, how do we teach them about risks along the way, and help them do this ethically and safely?  I'm super excited about this project and others like it that are trying to tackle these hard questions, and doing it with young children, rather than limiting kids' access. 💻 😄 

For this suggestion, I was thinking that the primary audience here is first the CS teachers, volunteers, or parents who would be introducing this to students, to help make sure they were aware of these issues so they can decide for themselves what to do.  Showing something simpler to young people seems even better!  Especially if it's understandable and not wall-of-legal-text.

To brainstorm, I remembered how Scratch cues young people not to use their real names in the sign-in flow.  This seems like a good balance maybe:

<img width="400" alt="Screen Shot 2019-03-21 at 12 53 55 PM" src="https://user-images.githubusercontent.com/1056957/54770136-0aa3bc80-4bd9-11e9-8557-766fbc6c24fb.png">

From there I tried to make something subtle to the UI at the point where children are deciding what to enter as data, but that is also direct.  This uses the example from Twitter earlier, where students were entering text messages to train a model that could tell who they were talking to in their family, so the buckets are "mom" and "sister":

<img width="400" src="https://user-images.githubusercontent.com/1056957/54772812-923ffa00-4bde-11e9-9d4b-0f9864023d0f.png"/>

Another iteration might check the text for common things in the browser before it travels over the network (eg, names, birthdates, other potentially personally identifiable information) and warn:

<img width="400" src="https://user-images.githubusercontent.com/1056957/54772803-8fdda000-4bde-11e9-9d75-cc892aefeede.png" />


It's hard to find the balance between simple, clean and low-friction learning, while also helping folks who are new to machine learning to learn about how to do it right in terms of privacy and ethics, especially with young children.

I'm super happy to keep brainstorming with you on this too, or to pitch in with working on this over time, and help with finding whatever you think the right balance is for your project.  The simple UX is so great, and I'm excited to try out more ways of using this with kids.  Thanks again for sharing your awesome work in the open! 👍
